[API 사이트]
* https://developer.pubg.com/

[참고자료]

1. https://velog.io/@tolerance/%EB%8D%B0%EC%9D%B4%ED%84%B0-%EB%B6%84%EC%84%9D%EC%9C%BC%EB%A1%9C-%EB%B0%B0%EA%B7%B8-PCS3-ASIA-%ED%95%9C%EA%B5%AD-2%EB%93%B1%ED%8C%80-%EC%98%88%EC%B8%A1%ED%95%9C-%EC%8D%B0-2

2. https://github.com/dataitgirls4/team_5/blob/main/raw-data/API-preface.py

3. https://right1203.tistory.com/2

In [ ]:
import requests
import json
import pandas as pd

* API key 값

In [ ]:
key = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJqdGkiOiI2NzVkMjJkMC1jNjE0LTAxM2ItZmJjZS0yZTE3NmZkNTJmNDkiLCJpc3MiOiJnYW1lbG9ja2VyIiwiaWF0IjoxNjgyNDgxNDU2LCJwdWIiOiJibHVlaG9sZSIsInRpdGxlIjoicHViZyIsImFwcCI6Ind3Y2RfcHJvamVjdCJ9.UYmIP4UeGG870rzOUgJOEPBHe3WTtqB0YbQWvEcrwdE'

# 선수 정보 API

In [ ]:
player = 'GEN_Pio'
url = f"https://api.pubg.com/shards/steam/players?filter[playerNames]={player}" 

header = {
  "Authorization": key,
  "Accept": "application/vnd.api+json"
}

r = requests.get(url, headers=header)
json = r.json()

json

{'data': [{'type': 'player',
   'id': 'account.e57c514acd19440bbc8a52233e482d93',
   'attributes': {'name': 'GEN_Pio',
    'stats': None,
    'titleId': 'pubg',
    'shardId': 'steam',
    'patchVersion': ''},
   'relationships': {'matches': {'data': [{'type': 'match',
       'id': 'd521ea8c-815c-4be8-bbae-ec877e15a0c2'},
      {'type': 'match', 'id': '82c0d71d-c069-4f17-931c-09c65000b8b0'},
      {'type': 'match', 'id': 'e2dd641a-6827-49f9-b5d6-d47fa8dd00d1'},
      {'type': 'match', 'id': '43867e57-597e-4361-89ab-9ec84c76fc68'},
      {'type': 'match', 'id': '523cf956-8756-49b9-b9c0-4ca09f33c8e5'},
      {'type': 'match', 'id': '144835e0-3d0c-4691-a19a-eeb111a2311f'},
      {'type': 'match', 'id': '5d01624e-c138-4f58-8e7f-3cf563038acc'},
      {'type': 'match', 'id': '6556baff-5270-4b60-acfa-9743c7e13680'},
      {'type': 'match', 'id': 'fc2d9982-4e74-4bbb-a712-45c2478852bb'},
      {'type': 'match', 'id': '630710c5-9fc9-4c09-9a72-52500211daf8'},
      {'type': 'match', 'id': '28099e

In [ ]:
json.keys()

dict_keys(['data', 'links', 'meta'])

In [ ]:
json['data'][0].keys()

dict_keys(['type', 'id', 'attributes', 'relationships', 'links'])

In [ ]:
match_data = json['data'][0]['relationships']['matches']['data']
match_data

[{'type': 'match', 'id': 'd521ea8c-815c-4be8-bbae-ec877e15a0c2'},
 {'type': 'match', 'id': '82c0d71d-c069-4f17-931c-09c65000b8b0'},
 {'type': 'match', 'id': 'e2dd641a-6827-49f9-b5d6-d47fa8dd00d1'},
 {'type': 'match', 'id': '43867e57-597e-4361-89ab-9ec84c76fc68'},
 {'type': 'match', 'id': '523cf956-8756-49b9-b9c0-4ca09f33c8e5'},
 {'type': 'match', 'id': '144835e0-3d0c-4691-a19a-eeb111a2311f'},
 {'type': 'match', 'id': '5d01624e-c138-4f58-8e7f-3cf563038acc'},
 {'type': 'match', 'id': '6556baff-5270-4b60-acfa-9743c7e13680'},
 {'type': 'match', 'id': 'fc2d9982-4e74-4bbb-a712-45c2478852bb'},
 {'type': 'match', 'id': '630710c5-9fc9-4c09-9a72-52500211daf8'},
 {'type': 'match', 'id': '28099e5b-a456-4cfc-9a60-488145e93086'},
 {'type': 'match', 'id': '6236c3fb-4703-44dc-a7c1-535a310eb31a'},
 {'type': 'match', 'id': '1117ef31-192d-432d-b6e6-95011ad3b30a'},
 {'type': 'match', 'id': '583f918f-10dd-4048-abc3-e8a6ba75b476'},
 {'type': 'match', 'id': 'd27d3036-bedb-45d8-b4eb-4db433cec28d'},
 {'type': 

In [ ]:
# 선수가 참여한 match id만 저장하기
df_match = pd.DataFrame(match_data)
df_match

type                                    id
0   match  d521ea8c-815c-4be8-bbae-ec877e15a0c2
1   match  82c0d71d-c069-4f17-931c-09c65000b8b0
2   match  e2dd641a-6827-49f9-b5d6-d47fa8dd00d1
3   match  43867e57-597e-4361-89ab-9ec84c76fc68
4   match  523cf956-8756-49b9-b9c0-4ca09f33c8e5
5   match  144835e0-3d0c-4691-a19a-eeb111a2311f
6   match  5d01624e-c138-4f58-8e7f-3cf563038acc
7   match  6556baff-5270-4b60-acfa-9743c7e13680
8   match  fc2d9982-4e74-4bbb-a712-45c2478852bb
9   match  630710c5-9fc9-4c09-9a72-52500211daf8
10  match  28099e5b-a456-4cfc-9a60-488145e93086
11  match  6236c3fb-4703-44dc-a7c1-535a310eb31a
12  match  1117ef31-192d-432d-b6e6-95011ad3b30a
13  match  583f918f-10dd-4048-abc3-e8a6ba75b476
14  match  d27d3036-bedb-45d8-b4eb-4db433cec28d
15  match  1f95a373-3e31-4b80-a7a9-abc74b219bfc
16  match  38357a2b-1f39-45d1-8c00-a23abbdf7025
17  match  a4bc9102-88d1-426e-9c95-937c3ae9b672
18  match  ff9c02ca-bb1d-43e7-9d8f-95e270bd65a9
19  match  b49e6eac-f257-461e-9794-3ac0e1d39b38
20  match  9d077032-12da-4c7f-827e-f0d282af7860
21  match  aee91e3a-0455-4915-aeac-2db1123e624e
22  match  f940e321-a7d1-4076-b10c-bf85aa4c66fd
23  match  88f765db-8cb9-4895-bf0e-dafb9c87bb15
24  match  2c580bce-edd8-491c-b7f4-b49b5d34e899
25  match  05365b86-2964-434f-b64c-142a9d37d2f8
26  match  ca33e3be-639e-4bbf-90c5-eb402068894c
27  match  93bedcc9-ab8b-4d11-a369-ab43761ad6cd
28  match  96f1085d-77d1-49dd-a751-ca73e7d85300
29  match  e37a0c03-5613-45ce-9b4d-9e6a278819bb
30  match  89789849-5ede-4e58-859a-8e6cfefff982
31  match  4a8aeb12-7d97-4569-a3d6-4261af27348d
32  match  0fcaeb42-67ef-4789-8070-e2ad847cf0ac
33  match  83939fca-5e75-4583-8aeb-afc57a5cef71
34  match  71b0e054-da05-4229-928d-8c01f8864942
35  match  00e4d27b-9bd8-4f90-b2f2-882196ace8bd
36  match  1b560766-acd0-4455-962d-ae1c0cd89a20
37  match  bc82c562-11a4-4d76-a830-3382ee3c053c
38  match  fff61220-b184-4bab-9c9c-138275e46399
39  match  62b4e09f-d84a-44de-bbe0-625412abbd19
40  match  9f5295da-efb8-4224-944d-1176efb87a6b
41  match  feee60f9-bdc4-42d9-b288-ae078a9a8de1
42  match  48c76a54-93b1-4c4b-ad33-e0f461314dbd
43  match  65064ccf-d0c2-4544-b1e3-4fade02c6762
44  match  67c026ff-9950-4952-86eb-27901af8a5fc
45  match  40fba429-09a4-47d1-a0e2-c780d8e1767f
46  match  4907d255-0e14-45ca-b543-c6b3d30e664c
47  match  347e8793-2d43-427c-a4dd-15dc27c1bf05
48  match  3bcb9567-1103-4cd5-8fc7-221b67816556
49  match  5e2bf1a8-e908-400d-b20f-8ce47cf51af2
50  match  b2059d70-40f7-4c62-923b-41b6f754ef78
51  match  2cd89491-897a-4c97-a971-14883c4587f4
52  match  7a6d9a7b-a36a-403c-9bd2-e38febfd1c6f
53  match  2e928ccb-b3ef-4820-9ff6-608ad112a38f
54  match  e231abd1-af21-4271-9813-b558b2edb813
55  match  e81ab7ae-0e65-43e9-a499-2a5bc6e3f8d5
56  match  c5fbbfbd-ae22-4f93-9989-aef187c5e25d
57  match  78771e17-0db6-4a01-b65e-ca8459885bfc
58  match  5cef868e-772c-499b-b4bd-4ef4503f01a4
59  match  53bd95a7-4fe9-433b-9d4d-0c7183bd2a37

In [ ]:
id = df_match.iloc[0, 1]
id

'd521ea8c-815c-4be8-bbae-ec877e15a0c2'

# 토너먼트 리스트 

In [ ]:
url_tnm = 'https://api.pubg.com/tournaments'
r_tnm = requests.get(url_tnm, headers=header)
tnm_json = r_tnm.json()
tnm_json

{'data': [{'type': 'tournament',
   'id': 'as-pgs1gs',
   'attributes': {'createdAt': '2023-04-25T12:44:50Z'}},
  {'type': 'tournament',
   'id': 'eu-neas23',
   'attributes': {'createdAt': '2023-04-14T19:35:54Z'}},
  {'type': 'tournament',
   'id': 'sea-leo32',
   'attributes': {'createdAt': '2023-04-03T11:39:09Z'}},
  {'type': 'tournament',
   'id': 'kr-pkp1gf',
   'attributes': {'createdAt': '2023-03-27T11:55:31Z'}},
  {'type': 'tournament',
   'id': 'am-pas1',
   'attributes': {'createdAt': '2023-03-24T23:39:38Z'}},
  {'type': 'tournament',
   'id': 'kr-pkp1w4',
   'attributes': {'createdAt': '2023-03-20T10:24:54Z'}},
  {'type': 'tournament',
   'id': 'eu-trp23',
   'attributes': {'createdAt': '2023-03-17T12:03:30Z'}},
  {'type': 'tournament',
   'id': 'kr-pkp1w3',
   'attributes': {'createdAt': '2023-03-13T10:29:56Z'}},
  {'type': 'tournament',
   'id': 'na-pas1rp',
   'attributes': {'createdAt': '2023-03-12T00:34:12Z'}},
  {'type': 'tournament',
   'id': 'sa-pas1rp',
   'attribut

# 특정 토너먼트의 매치 id 불러오기

In [ ]:
tnm = 'kr-pkp1gf'
url_league = f"https://api.pubg.com/tournaments/{tnm}"
r_league = requests.get(url_league, headers=header)
league_json = r_league.json()
league_json

{'data': {'type': 'tournament',
  'id': 'kr-pkp1gf',
  'relationships': {'matches': {'data': [{'type': 'match',
      'id': '1073683e-4a30-45a1-a0a8-ac48c72d6784'},
     {'type': 'match', 'id': '79e592fc-9abe-4f54-ae77-bf1abc283613'},
     {'type': 'match', 'id': 'fe7adb64-005a-4e37-b9c5-d7faf9429950'},
     {'type': 'match', 'id': 'eca683cc-9dd4-456c-a4d7-511dc2d32442'},
     {'type': 'match', 'id': '031e574f-7533-4ce0-a76e-9fb18beb736d'},
     {'type': 'match', 'id': 'ba9933e2-1add-4fe3-81f2-eaca4a2cdfb7'},
     {'type': 'match', 'id': '4c92d1eb-714c-420e-a424-da3a1daf0b48'},
     {'type': 'match', 'id': '36dc9e56-0578-4c2f-95e3-13b7d6fa5366'},
     {'type': 'match', 'id': '1e075b63-adef-496b-8790-b76acd259f1f'},
     {'type': 'match', 'id': 'e2127bb7-3a6b-4a53-b8ee-41de6607f07b'},
     {'type': 'match', 'id': '72ee15dc-737f-4ece-90f3-d0b9a31f8e8d'},
     {'type': 'match', 'id': '6cbb4e53-6048-4cfb-8bd2-a5ffdc0251ff'},
     {'type': 'match', 'id': '82029ce0-20b2-44d1-aeca-a26f47fa487

* 한 토너먼트에 총 20개 매치가 있음을 확인.

In [ ]:
len(league_json['included'])

20

In [ ]:
len(league_json['data']['relationships']['matches']['data'])

20

# 가장 최근 한국 리그 매치 id로 매치 player 정보 받아오기

In [ ]:
id = '1073683e-4a30-45a1-a0a8-ac48c72d6784'

In [ ]:
url_mat = f'https://api.pubg.com/shards/tournaments/matches/{id}'
r_mat = requests.get(url_mat, headers=header)
mat_json = r_mat.json()
mat_json

{'data': {'type': 'match',
  'id': '1073683e-4a30-45a1-a0a8-ac48c72d6784',
  'attributes': {'duration': 1814,
   'shardId': 'tournament',
   'isCustomMatch': True,
   'titleId': 'bluehole-pubg',
   'tags': None,
   'mapName': 'Desert_Main',
   'matchType': '',
   'seasonState': 'progress',
   'createdAt': '2023-03-31T12:52:34Z',
   'stats': None,
   'gameMode': 'esports-squad-fpp'},
  'relationships': {'rosters': {'data': [{'type': 'roster',
      'id': 'afc08730-d826-4a9f-acf7-830cdccaf670'},
     {'type': 'roster', 'id': '16594251-8dc6-4b37-abd9-4eb47ae61d98'},
     {'type': 'roster', 'id': '59a8795a-1b05-4e4b-a00e-c4cd86bc29b1'},
     {'type': 'roster', 'id': 'fc0718be-358e-400a-b6a7-5e7f10f7d581'},
     {'type': 'roster', 'id': '1538fd3f-d1ff-45c6-93e8-7679a40a2f9a'},
     {'type': 'roster', 'id': '8471ef24-cebe-4aae-97f8-eb5607887945'},
     {'type': 'roster', 'id': 'a8c1b849-0177-4ca4-8646-dd25c22a3e03'},
     {'type': 'roster', 'id': '63642ccf-56ba-4369-ae50-fd77087195e9'},
    

In [ ]:
mat_json.keys()

dict_keys(['data', 'included', 'links', 'meta'])

* 선수의 경기 정보가 들어있는 딕셔너리 찾기

In [ ]:
mat_player = mat_json['included'][0]['attributes']['stats']
mat_player

{'DBNOs': 0,
 'assists': 1,
 'boosts': 1,
 'damageDealt': 132.61337,
 'deathType': 'byplayer',
 'headshotKills': 0,
 'heals': 1,
 'killPlace': 64,
 'killStreaks': 0,
 'kills': 0,
 'longestKill': 0,
 'name': 'GnG_BeaN',
 'playerId': 'account.28cbaadd08034c538773788683a003f5',
 'revives': 0,
 'rideDistance': 3170.1406,
 'roadKills': 0,
 'swimDistance': 0,
 'teamKills': 0,
 'timeSurvived': 565,
 'vehicleDestroys': 0,
 'walkDistance': 614.27637,
 'weaponsAcquired': 6,
 'winPlace': 16}

In [ ]:
mat_player.keys()

dict_keys(['DBNOs', 'assists', 'boosts', 'damageDealt', 'deathType', 'headshotKills', 'heals', 'killPlace', 'killStreaks', 'kills', 'longestKill', 'name', 'playerId', 'revives', 'rideDistance', 'roadKills', 'swimDistance', 'teamKills', 'timeSurvived', 'vehicleDestroys', 'walkDistance', 'weaponsAcquired', 'winPlace'])

In [ ]:
pd.DataFrame(mat_player, index=[0])

DBNOs  assists  boosts  damageDealt deathType  headshotKills  heals  \
0      0        1       1    132.61337  byplayer              0      1   

   killPlace  killStreaks  kills  ...  revives rideDistance roadKills  \
0         64            0      0  ...        0    3170.1406         0   

   swimDistance  teamKills  timeSurvived  vehicleDestroys  walkDistance  \
0             0          0           565                0     614.27637   

   weaponsAcquired  winPlace  
0                6        16  

[1 rows x 23 columns]

* 16개팀 * 팀당 4선수 -> 64개 정보가 나와야 하는데 81개가 나옴
* 중간에 랭크 정보 같은 다른 내용도 섞여있음
* 데이터프레임으로 바꾸는 방법 찾기

In [ ]:
len(mat_json['included'])

81

In [ ]:
81 -64

17

In [ ]:
mat_json['included'][80]

{'type': 'roster',
 'id': '4982b4da-76fd-4dac-a96a-f2299b6f4c31',
 'attributes': {'won': 'false',
  'shardId': 'tournament',
  'stats': {'rank': 2, 'teamId': 12}},
 'relationships': {'team': {'data': None},
  'participants': {'data': [{'type': 'participant',
     'id': '573af90e-709e-4bac-83ce-a3432afadb93'},
    {'type': 'participant', 'id': '9eda927e-4914-454d-83ce-97f3604da592'},
    {'type': 'participant', 'id': '468db08c-2c44-49fa-8906-2a54cd19a9bc'},
    {'type': 'participant', 'id': 'e0aa4fb5-2dd5-433a-a253-a8191436e319'}]}}}

In [ ]:
mat_json['included'][70]['attributes']

{'stats': {'rank': 16, 'teamId': 1}, 'won': 'false', 'shardId': 'tournament'}

In [ ]:
included = pd.DataFrame(mat_json['included'])
rosters = included[included['type'] == 'roster']
print(rosters.shape)
rosters

(16, 4)


type                                    id  \
6   roster  fc0718be-358e-400a-b6a7-5e7f10f7d581   
13  roster  1538fd3f-d1ff-45c6-93e8-7679a40a2f9a   
14  roster  1f8569ea-23e7-42aa-9629-fbfc9b287ce2   
20  roster  63642ccf-56ba-4369-ae50-fd77087195e9   
21  roster  6749e656-7d38-4a75-8ec1-96da24ce6792   
26  roster  c8472c50-b60c-4d1a-b1bd-8a7b723d0f7a   
39  roster  afc08730-d826-4a9f-acf7-830cdccaf670   
44  roster  32d07045-7db1-4474-9955-9592c1c2f91a   
46  roster  a8c1b849-0177-4ca4-8646-dd25c22a3e03   
47  roster  72bdaa87-e2ab-40b1-99ca-4bd43f5d6f6b   
50  roster  16594251-8dc6-4b37-abd9-4eb47ae61d98   
69  roster  59a8795a-1b05-4e4b-a00e-c4cd86bc29b1   
70  roster  8471ef24-cebe-4aae-97f8-eb5607887945   
77  roster  04e2b3c2-d655-4c84-9138-9f31dc12b90f   
78  roster  487d8dab-cad2-49df-a84d-db1e2a4e8e86   
80  roster  4982b4da-76fd-4dac-a96a-f2299b6f4c31   

                                           attributes  \
6   {'stats': {'rank': 1, 'teamId': 3}, 'won': 'tr...   
13  {'won': 'false', 'shardId': 'tournament', 'sta...   
14  {'shardId': 'tournament', 'stats': {'rank': 13...   
20  {'shardId': 'tournament', 'stats': {'rank': 6,...   
21  {'stats': {'rank': 14, 'teamId': 5}, 'won': 'f...   
26  {'won': 'false', 'shardId': 'tournament', 'sta...   
39  {'stats': {'rank': 8, 'teamId': 9}, 'won': 'fa...   
44  {'stats': {'rank': 4, 'teamId': 16}, 'won': 'f...   
46  {'stats': {'rank': 12, 'teamId': 8}, 'won': 'f...   
47  {'won': 'false', 'shardId': 'tournament', 'sta...   
50  {'stats': {'rank': 15, 'teamId': 6}, 'won': 'f...   
69  {'stats': {'rank': 9, 'teamId': 14}, 'won': 'f...   
70  {'stats': {'rank': 16, 'teamId': 1}, 'won': 'f...   
77  {'stats': {'rank': 3, 'teamId': 15}, 'won': 'f...   
78  {'stats': {'rank': 7, 'teamId': 13}, 'won': 'f...   
80  {'won': 'false', 'shardId': 'tournament', 'sta...   

                                        relationships  
6   {'team': {'data': None}, 'participants': {'dat...  
13  {'team': {'data': None}, 'participants': {'dat...  
14  {'team': {'data': None}, 'participants': {'dat...  
20  {'team': {'data': None}, 'participants': {'dat...  
21  {'team': {'data': None}, 'participants': {'dat...  
26  {'participants': {'data': [{'type': 'participa...  
39  {'team': {'data': None}, 'participants': {'dat...  
44  {'team': {'data': None}, 'participants': {'dat...  
46  {'team': {'data': None}, 'participants': {'dat...  
47  {'team': {'data': None}, 'participants': {'dat...  
50  {'team': {'data': None}, 'participants': {'dat...  
69  {'team': {'data': None}, 'participants': {'dat...  
70  {'team': {'data': None}, 'participants': {'dat...  
77  {'participants': {'data': [{'type': 'participa...  
78  {'team': {'data': None}, 'participants': {'dat...  
80  {'team': {'data': None}, 'participants': {'dat...

## 데이터프레임 만들기

* 'type': 'participant 

    => 64명의 선수 정보가 담겨있는 딕셔너리
* 'type': 'roster'

    => rank, teamId 등 정보가 담겨있는 딕셔너리

In [ ]:
# 64명 선수 정보만 데이터 프레임으로 저장하기
participant_indexes = []
for i, item in enumerate(mat_json['included']):
    if item.get('type') == 'participant':
        participant_indexes.append(i)

dfs = []
for i in participant_indexes:
    mat_player = mat_json['included'][i]['attributes']['stats']
    df_mat_player = pd.DataFrame(mat_player, index=[0])
    dfs.append(df_mat_player)

result_df = pd.concat(dfs, ignore_index=True)
result_df

DBNOs  assists  boosts  damageDealt deathType  headshotKills  heals  \
0       0        1       1   132.613370  byplayer              0      1   
1       0        0       1     0.000000  byplayer              0      0   
2       1        0       3   145.934230  byplayer              0      4   
3       3        0       8   380.775270  byplayer              0      2   
4       2        1       3   386.104950  byplayer              0      5   
..    ...      ...     ...          ...       ...            ...    ...   
59      0        1       1    51.660004  byplayer              0      0   
60      2        0       3   187.370470  byplayer              0      2   
61      0        0       7   215.967040  byplayer              0      4   
62      1        2       4   355.830350  byplayer              0      1   
63      5        2       7   715.662960     alive              2      3   

    killPlace  killStreaks  kills  ...  revives rideDistance roadKills  \
0          64            0      0  ...        0    3170.1406         0   
1          54            0      0  ...        0    5795.8027         0   
2          32            0      0  ...        0    8513.4140         0   
3           9            1      3  ...        2    5223.9460         0   
4          11            1      2  ...        0    8217.5630         0   
..        ...          ...    ...  ...      ...          ...       ...   
59         38            0      0  ...        0    7126.0894         0   
60          8            2      3  ...        0    4442.5240         0   
61         31            0      0  ...        0    2857.3955         0   
62          5            2      4  ...        0    7396.8680         0   
63          1            2      6  ...        0    3467.7490         0   

    swimDistance  teamKills  timeSurvived  vehicleDestroys  walkDistance  \
0              0          0           565                0     614.27637   
1              0          0          1031                0    1086.54860   
2              0          0          1384                0    1350.04810   
3              0          0          1459                0    1536.37660   
4              0          0          1495                0    2012.55540   
..           ...        ...           ...              ...           ...   
59             0          0           626                0     400.35043   
60             0          0          1482                1    1237.87020   
61             0          0          1609                0    1981.38660   
62             0          0          1495                0    2550.41140   
63             0          0          1814                0    2728.25120   

    weaponsAcquired  winPlace  
0                 6        16  
1                 4        12  
2                10         3  
3                10         5  
4                 3         4  
..              ...       ...  
59                6         7  
60                8         5  
61                4         2  
62               10         4  
63                8         1  

[64 rows x 23 columns]

### 칼럼명 파헤치기
[참고자료]
https://5quad.github.io/2019/11/29/AIXProject.html

- 'timeSurvived': 737 
=> 초 단위인 듯 (약 12분 정도 생존시간)

- 'rideDistance': 2327.0278,
=> 오토바이 또는 자동차 탄 거리

- 'walkDistance': 707.66125,
=> 걸어서 이동한 거리

- 'deathType': 'byplayer',
=> by player  

    => 자살한 경우가 아님 ㅋㅋㅋ
    
    => 혼자 죽는 경우 ex) 자기장 타 죽거나, 기절했는데 수류탄 떨어지거나

- 'killPlace': 60
=> 매치에서의 킬 횟수 순위

- 'winPlace': 15
=> 우승 순위...??

* killPlace 는 `매치에서의 킬 횟수 순위` 를 의미함.

 1부터 64까지 고유값이 있음.
 
 ⁉️ 의문점 
 
 kills 가 똑같이 0 인데 `매치에서의 킬 횟수 순위` 를 매기는 기준은 뭐지...? 그냥 랜덤인가...?

 내일 오전에 좀 더 고민하기

In [ ]:
result_df[result_df['kills'] == 0].sort_values(by='killPlace')

DBNOs  assists  boosts  damageDealt deathType  headshotKills  heals  \
20      1        2       5   181.531940  byplayer              0      2   
61      0        0       7   215.967040  byplayer              0      4   
2       1        0       3   145.934230  byplayer              0      4   
49      1        1       2   148.397640  byplayer              0      0   
16      0        1       0   112.135730  byplayer              0      1   
15      0        1       1    81.180010  byplayer              0      2   
41      0        0       0     0.000000  byplayer              0      0   
25      0        0       1    22.140001  byplayer              0      0   
59      0        1       1    51.660004  byplayer              0      0   
43      0        0       1     0.000000  byplayer              0      0   
12      0        0       4    76.066080  byplayer              0      0   
17      0        0       5    89.894700  byplayer              0      1   
32      0        0       5    35.479465  byplayer              0      0   
45      0        0       1     0.000000  byplayer              0      0   
5       0        0       3    51.660004    byzone              0      2   
13      0        0       1     0.000000  byplayer              0      0   
35      0        0       0    44.984398  byplayer              0      0   
37      0        0       2    62.730003  byplayer              0      0   
29      0        0       2     0.000000  byplayer              0      2   
58      0        0       1   126.900390  byplayer              0      1   
48      2        2       3   199.999980  byplayer              0      2   
55      1        0       1   100.000000  byplayer              0      0   
38      0        0       4    69.215660  byplayer              0      1   
36      0        0       4    84.120000  byplayer              0      0   
1       0        0       1     0.000000  byplayer              0      0   
18      0        0       2    27.329030  byplayer              0      0   
30      0        1       1    27.060001  byplayer              0      0   
47      0        0       2    41.580000  byplayer              0      0   
9       0        0       1    20.295000  byplayer              0      0   
33      0        0       2    95.641075  byplayer              0      0   
53      1        1       1   146.444720  byplayer              0      1   
57      0        0       1     0.000000  byplayer              0      0   
23      0        0       3    29.095825  byplayer              0      2   
8       0        0       0    43.050000  byplayer              0      0   
0       0        1       1   132.613370  byplayer              0      1   

    killPlace  killStreaks  kills  ...  revives rideDistance roadKills  \
20         30            0      0  ...        0    4286.9434         0   
61         31            0      0  ...        0    2857.3955         0   
2          32            0      0  ...        0    8513.4140         0   
49         33            0      0  ...        0    8661.4840         0   
16         34            0      0  ...        0    5393.2470         0   
15         35            0      0  ...        1    4612.0530         0   
41         36            0      0  ...        0    8461.3130         0   
25         37            0      0  ...        0   10043.1690         0   
59         38            0      0  ...        0    7126.0894         0   
43         39            0      0  ...        0    5559.3706         0   
12         40            0      0  ...        1    3496.6855         0   
17         41            0      0  ...        0    3906.7550         0   
32         42            0      0  ...        0    3892.6950         0   
45         43            0      0  ...        0    3749.0127         0   
5          44            0      0  ...        0    6741.3390         0   
13         45            0      0  ...        0    5096.7725         0   
35         46            0      0  ...       

In [ ]:
result_df.columns

Index(['DBNOs', 'assists', 'boosts', 'damageDealt', 'deathType',
       'headshotKills', 'heals', 'killPlace', 'killStreaks', 'kills',
       'longestKill', 'name', 'playerId', 'revives', 'rideDistance',
       'roadKills', 'swimDistance', 'teamKills', 'timeSurvived',
       'vehicleDestroys', 'walkDistance', 'weaponsAcquired', 'winPlace'],
      dtype='object')

In [ ]:
result_df['deathType'].unique()

array(['byplayer', 'byzone', 'alive'], dtype=object)

In [ ]:
sorted(list(result_df['killPlace'].unique()))

[1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 20,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28,
 29,
 30,
 31,
 32,
 33,
 34,
 35,
 36,
 37,
 38,
 39,
 40,
 41,
 42,
 43,
 44,
 45,
 46,
 47,
 48,
 49,
 50,
 51,
 52,
 53,
 54,
 55,
 56,
 57,
 58,
 59,
 60,
 61,
 62,
 63,
 64]

# 배그 패키지 불러오기

## (아직 테스트 해봐야함...ㅠㅠ)

1. chicken-dinner 패키지

In [ ]:
!pip install click==7.1.2
!pip install flask==2.0.1

In [ ]:
!pip install chicken-dinner

In [ ]:
!pip install chicken-dinner[visual]

In [ ]:
from chicken_dinner.pubgapi import PUBGCore

api_key = key #"your_api_key"
pubgcore = PUBGCore(api_key, "pc-na")
shroud = pubgcore.players("player_names", "shroud")
print(shroud)

{'data': [{'type': 'player', 'id': 'account.d50fdc18fcad49c691d38466bed6f8fd', 'attributes': {'name': 'shroud', 'stats': None, 'titleId': 'pubg', 'shardId': 'pc-na', 'patchVersion': ''}, 'relationships': {'assets': {'data': []}, 'matches': {'data': []}}, 'links': {'self': 'https://api.pubg.com/shards/pc-na/players/account.d50fdc18fcad49c691d38466bed6f8fd', 'schema': ''}}], 'links': {'self': 'https://api.playbattlegrounds.com/shards/pc-na/players?filter[playerNames]=shroud'}, 'meta': {}}


In [ ]:
from chicken_dinner.pubgapi import PUBG

api_key = key #"your_api_key"
pubg = PUBG(api_key, "pc-na")
shroud = pubg.players_from_names("shroud")[0]
shroud_season = shroud.get_current_season()
squad_fpp_stats = shroud_season.game_mode_stats("squad", "fpp")
print(squad_fpp_stats)

{'assists': 0, 'boosts': 0, 'dbnos': 0, 'daily_kills': 0, 'daily_wins': 0, 'damage_dealt': 0, 'days': 0, 'headshot_kills': 0, 'heals': 0, 'kill_points': 0, 'kills': 0, 'longest_kill': 0, 'longest_time_survived': 0, 'losses': 0, 'max_kill_streaks': 0, 'most_survival_time': 0, 'rank_points': 0, 'rank_points_title': '', 'revives': 0, 'ride_distance': 0, 'road_kills': 0, 'round_most_kills': 0, 'rounds_played': 0, 'suicides': 0, 'swim_distance': 0, 'team_kills': 0, 'time_survived': 0, 'top_10s': 0, 'vehicle_destroys': 0, 'walk_distance': 0, 'weapons_acquired': 0, 'weekly_kills': 0, 'weekly_wins': 0, 'win_points': 0, 'wins': 0}


In [ ]:
from chicken_dinner.pubgapi import PUBG

api_key = key #"your_api_key"
pubg = PUBG(api_key, "pc-na")
shroud = pubg.players_from_names("shroud")[0]
shroud

In [ ]:
shroud.match_ids

[]

In [ ]:
recent_match_id = shroud.match_ids[0]
recent_match = pubg.match(recent_match_id)
recent_match_telemetry = recent_match.get_telemetry()
recent_match_telemetry.playback_animation("recent_match.html")

2. pubg-python 패키지

In [ ]:
!pip install pubg-python

In [ ]:
from pubg_python import PUBG, Shard
from pubg_python.domain.telemetry.events import LogMatchEnd

In [ ]:
api = PUBG(api_key=key, shard=Shard.PC_KRJP)

# 프로 선수의 닉네임을 지정합니다.
player_name = player

# 매치 정보를 가져오기 위한 필터를 지정합니다.
filter = {'playerNames': [player_name]}

# 매치 리스트를 가져옵니다.
matches = api.matches(filter=filter)

# 최근 5개의 매치에서 매치 정보를 출력합니다.
for match in matches[:5]:
    # 매치 정보를 가져옵니다.
    telemetry = api.telemetry(match.id)

    # 매치 정보에서 매치 종료 이벤트를 찾습니다.
    match_end_event = telemetry.find_first(LogMatchEnd)

    # 매치 종료 이벤트에서 우승자 정보를 가져옵니다.
    winner = match_end_event.winners[0]

    print(f'Match ID: {match.id}, Map: {match.map_name}, Mode: {match.game_mode}, Winner: {winner.name}')


In [ ]:
matches

In [ ]:
from pubg_python import PUBG, Shard

api = PUBG(key, Shard.PC_NA)

In [ ]:
sample = api.samples().get()
for match in sample.matches:
    print(match.id)

In [ ]:
players = api.players().filter(player_names=['Pio'])

for player in players:
    player_id = player.id

In [ ]:
season_data = api.seasons('division.bro.official.2018-09' , player_id='epickitten').get()

In [ ]:
season_data = api.seasons(
    'lifetime' , player_id='epickitten').get()

In [ ]:
players = api.players().filter(player_names=['epickitten'])
player = players[0]

player.matches()